### PR TITLE
Modernize the packaging for ipywidgets

### PR DESCRIPTION
**Pull request by jasongrout**
This tries to update our packaging to the newest recommended standards, using setup.cfg for as much as possible.

CC @ibdafna

### CODE REVIEW

**Comment by jasongrout** [comment]
Converting to draft since I want to tweak the jupyterlab_widgets packaging a bit as well

**Comment by jasongrout** [comment]
> Converting to draft since I want to tweak the jupyterlab_widgets packaging a bit as well

Ready for others to look at it now.

**Review by vidartf — APPROVED** [review]
This seems good to me, and was certainly instructive. We should of course make sure we test the different dist ways during beta/rc, but I cannot see any misses.

File: docs/source/dev_release.md [hunk 1]
```diff
@@ -59,8 +59,8 @@ Lerna will prompt you for version numbers for each of the changed npm packages i
 
 Go into the `jupyterlab_widgets` directory. Change `jupyterlab_widgets/_version.py` to reflect the new version number.
 ```
-python setup.py sdist bdist_wheel
-twine check dist/*
+pip install build
+python -m build
```
Comment (jtpio) [inline docs/source/dev_release.md:63]: Should we also update the packaging workflow to use these new commands?

https://github.com/jupyter-widgets/ipywidgets/blob/3108308d93c423845901500b108f45b4d7e9fc5b/.github/workflows/packaging.yml#L56

Comment (jasongrout) [reply docs/source/dev_release.md:63]: Good point. Done!

**Comment by jasongrout** [comment]
@jtpio - the packaging tests are now passing. Once someone merges this, I'll release a new set of alphas so we can test more comprehensively.

**Comment by jasongrout** [comment]
Always a good sign :)

Screen Shot 2021-02-19 at 9 51 24 AM

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jtpio** [comment]
> the packaging tests are now passing. Once someone merges this, I'll release a new set of alphas so we can test more comprehensively.

Alright, then let's merge and adjust the setup and steps if needed :+1:

**Comment by jasongrout** [comment]
> Alright, then let's merge and adjust the setup and steps if needed 👍

Released as ipywidgets 8.0.0a4, widgetsnbextension 4.0.0a2, jupyterlab_widgets 2.0.0a1

I did a test install in a clean environment and ran a minimal example in both jlab3 and the notebook, and both seemed to work.

**Comment by jtpio** [comment]
Thanks Jason!

> Released as ipywidgets 8.0.0a4, widgetsnbextension 4.0.0a2, jupyterlab_widgets 2.0.0a1

I can also confirmed these alpha releases work fine locally in a new environment. And [on Binder](https://mybinder.org/v2/gist/jtpio/d4fd0ba2e0945fff407097635a8dc209/master?urlpath=/lab/tree/widgets_list.ipynb) with this [gist](https://gist.github.com/jtpio/d4fd0ba2e0945fff407097635a8dc209) too.

**Pull request merged**